### PR TITLE
move some vector allocations into initialization functions

### PR DESCRIPTION
**Pull request by boyceg**

### CODE REVIEW

**Comment by amneetb** [comment]
@nishantnangia : Can you test this branch locally for multiphase solver?

**Comment by nishantnangia** [comment]
I checked some cases earlier this evening and nothing seemed to
break/change — so I approved the PR.

File: src/navier_stokes/VCStaggeredStokesOperator.cpp [hunk 1]
```diff
@@ -85,9 +85,6 @@ VCStaggeredStokesOperator::apply(SAMRAIVectorReal<NDIM, double>& x, SAMRAIVector
 {
     IBAMR_TIMER_START(t_apply);
 
-    // Allocate scratch data.
-    d_x->allocateVectorData();
-
```
Comment (drwells) [inline src/navier_stokes/VCStaggeredStokesOperator.cpp]: It looks like you forgot one.

**Comment by boyceg** [comment]
> @drwells requested changes on this pull request.
> We need to make a similar fix to VCStaggeredStokesOperator (its missing the allocate now, AFAICT)
> 

Those happen in the base class I think?

**Comment by drwells** [comment]
Whoops - you're absolutely right.

**Comment by drwells** [comment]
@nishantnangia and I are both happy so lets merge!

**Comment by drwells** [comment]
@marshallrdavey This PR might actually help with the increase in memory usage problem. If its easy to do, could you use `master` for the next round of heart simulations and see if there is any change in the output from `getrusage`?

File: src/navier_stokes/VCStaggeredStokesOperator.cpp (hunk 1 above)
Comment (boyceg) [reply src/navier_stokes/VCStaggeredStokesOperator.cpp]: Just to close the loop --- allocation/deallocation is being handled in the base class.

**Pull request merged**